dcp/DcpHandler.go

```diff
@@ -654,3 +661,34 @@ func removeKVSubsetFromXattr(xattr []byte, size int, xattrSize uint32, xattrIter
 	trimmedXattrPlusBody, _ = xattrComposer.FinishAndAppendDocValue(bodyWithoutXattr)
 	return trimmedXattrPlusBody, KVsToBeExcluded, nil
 }
+
+func GetImportCasAndPrevFromMou(mou []byte) (importCas uint64, pRev uint64, err error) {
+	if xdcrBase.Equals(mou, xdcrBase.EmptyJsonObject) {
+		return
+	}
+	newMou := make([]byte, len(mou))
+	removedFromMou := make(map[string][]byte)
+	_, _, _, err = gojsonsm.MatchAndRemoveItemsFromJsonObject(mou, xdcrBase.MouXattrValuesForCR, newMou, removedFromMou)
+	if err != nil {
+		return
+	}
+	xattrImportCas, foundImportCas := removedFromMou[xdcrBase.IMPORTCAS]
+	xattrPRev, foundPRev := removedFromMou[xdcrBase.PREVIOUSREV]
+	if foundImportCas && xattrImportCas != nil {
+		// Remove the start/end quotes before converting it to uint64
+		xattrLen := len(xattrImportCas)
+		importCas, err = xdcrBase.HexLittleEndianToUint64(xattrImportCas[1 : xattrLen-1])
+		if err != nil {
+			return
+		}
+	}
+	if foundPRev && xattrPRev != nil {
+		// Remove the start/end quotes before converting it to uint64
+		xattrLen := len(xattrPRev)
+		pRev, err = strconv.ParseUint(string(xattrPRev[1:xattrLen-1]), 10, 64)
+		if err != nil {
+			return
+		}
+	}
+	return
+}
```

Comment:
I see a lot of duplicate logic w.r.t. the XDCR code itself here:

https://github.com/couchbase/goxdcr/blob/5626def8be4cf63ae242b549242acdf9f483c159/crMeta/crMetadata.go#L928-L970

Is there anyway to make `GetImportCasAndPrevFromMou` a part of the goxdcr code and to be used as part of `crMetadata.go`? Then, simply use that function here in the differ instead of re-implementing here?

Differ should not have the logics of how to parse mou ... goxdcr should be the source of truth for parsing mou and differ simply calls the correct library on it

Reply:
Yeah I'll raise a ticket and I'll export this part of the code as a function in goxdcr.